godtools/App/Share/Data/UserAuthentication/AuthenticationProviders/AuthenticationProviderAccessToken.swift

```diff
@@ -8,8 +8,27 @@
 
 import Foundation
 
-struct AuthenticationProviderAccessToken {
+enum AuthenticationProviderAccessToken {
     
-    let provider: AuthenticationProviderType
-    let tokenString: String
+    case apple(idToken: String, givenName: String, familyName: String)
+    case facebook(accessToken: String)
+    case google(idToken: String)
+}
```

Comment:
Hey @rachaelblue this enum is great, we do have these distinct parameters being sent to the mobile content api.

One thing I like to ask myself when deciding what best goes where is to think about the dependency.  For example, since MobileContentAuthTokenAPI.swift now depends on AuthenticationProviderAccessToken.swift, what would happen if we created a separate module/namespace that MobileContentAuthTokenAPI.swift lived in.  Would AuthenticationProviderAccessToken.swift make sense living in that module?  Here I would say yes, because it contains the specific requirements that the MobileContentAuthTokenAPI is sending in the request body.  

I would also say that the original tokenString in struct AuthenticationProviderAccessToken didn't make sense because our authentication providers really could provide an accessToken, idToken, and a refreshToken according to OIDC (https://developer.okta.com/blog/2017/07/25/oidc-primer-part-1#all-about-tokens).

With that all said, I don't see any issues leaving this as is.  Both of these classes are living in the same project together and can depend on this enum.  

I should also make some updates to  ```AuthenticationProviderInterface.swift``` to follow more closely to the OIDC spec.  It should be providing accessToken, idToken, refreshToken, user name etc.

Reply:
That makes sense, thanks for explaining!